.github/workflows/test-smoketests.yml

```diff
@@ -51,6 +51,12 @@ jobs:
     - name: decorator smoke test
       run: python test/smoketest_profile_decorator.py
 
+      # NOTE: this is a regression test for signals not being
+      # restarted properly. It checks whether the program 
+      # halts or runs forever, not for correctness of timing.
+    - name: signal smoketest
+      run: python test/smoketest.py test/signal_test.py
```

Comment:
This is a bad failure case, plus we have a regression test suite in `tests/`. Make it a pytest and put it in that directory so it gets run in that workflow, but with a timeout for failure.

Reply:
I think the failure case is the right one-- the behaviors this patch was in response to were the signal not being delivered for whatever reason and this test checks for progress. 

I'll put it in the `tests` workflow, I actually think it makes sense as a `task` with a timeout rather than as a pytest though, unless you wanted to start making pytests that spawn new processes (which it doesn't seem like we do at the moment, the pytest suite seems to be mostly unit tests). I do think it's structurally more similar to the smoketests, since we're just checking "are the signals delivered and does it create a profile with any data in it", but that's your call

Reply:
I mean that the CI will get hung up - no idea what the timeout is for that thing - so the latter suggestion works.

Reply:
The timeout of the CI is 15 minutes and I put in a timeout of 30 seconds for the task I created, where would the hangup happen?

Reply:
Also [the smoketests worked fine](https://github.com/plasma-umass/scalene/actions/runs/17362592575/job/49284657107?pr=946), I could make sure the timeout works as expected but I trust the `timeout-minutes` directive to do the right thing